File: nagasenacs/src/Nagasena/Sax/SAXTransmogrifier.cs
namespace Nagasena.Sax {

  using ContentHandler = Org.System.Xml.Sax.IContentHandler;
  using LexicalHandler = Org.System.Xml.Sax.ILexicalHandler;

  using GrammarCache = Nagasena.Proc.Grammars.GrammarCache;

  /// <summary>
  /// Applications can directly feed SAX events into a transmogrifier through SAXTransmogrifier.
  /// </summary>
  public interface SAXTransmogrifier : ContentHandler, LexicalHandler, BinaryDataHandler {

    /// <summary>
    /// Returns the GrammarCache that is in use by this SAXTransmogrifier. </summary>
    /// <returns> a GrammarCache </returns>
    GrammarCache GrammarCache { get; }

  }

}

File: nagasena/src/org/openexi/sax/SAXTransmogrifier.java
package org.openexi.sax;

import org.xml.sax.ContentHandler;
import org.xml.sax.ext.LexicalHandler;

import org.openexi.proc.grammars.GrammarCache;

/**
 * Applications can directly feed SAX events into a transmogrifier through SAXTransmogrifier.
 */
public interface SAXTransmogrifier extends ContentHandler, LexicalHandler, BinaryDataHandler {

  /**
   * Returns the GrammarCache that is in use by this SAXTransmogrifier.
   * @return a GrammarCache
   */
  public GrammarCache getGrammarCache();
  
}
